src/loan_form/LoanForm.jsx

```diff
@@ -29,14 +30,68 @@ export default class LoanForm extends React.Component {
     this.setState({ loans: newLoans })
   }
   
+  handleChange(id, name, value) {
+    const newLoans = this.state.loans
+
+    for (var i = 0; i < this.state.loans.length; i++) {
+      if (this.state.loans[i]['id'] == parseInt(id)) {
+        newLoans[i][camelCase(name)] = value
+        newLoans[i].monthsToPayOffLoan = this.findLoanLength(newLoans[i])
+      }
+    }
+
+    this.setState({ loans: newLoans })
+  }
+
+  daysInYear(year) {
+    return 365
+  }
+
+  daysInMonth(month) {
+    return 30.417
+  }
+
+  effectiveMonthlyInterest(amountOwed, yearlyInterestRate, month, year) {
+    return amountOwed * (((1 + yearlyInterestRate / this.daysInYear(year)) ** this.daysInMonth(month)) - 1)
+  }
+
+  findLoanLength({amountOwed, interestRate, monthlyPayment}) {
+    
+    let tempAmountOwed = parseFloat(amountOwed)
+    let interest = this.effectiveMonthlyInterest(tempAmountOwed, interestRate)
+    let monthsToPayOffLoan = 0
+
+    if (monthlyPayment <= interest && amountOwed > 0) {
+      monthsToPayOffLoan = Infinity
+    } else {
+      while (tempAmountOwed > 0) {
+        monthsToPayOffLoan += 1
+        interest = this.effectiveMonthlyInterest(tempAmountOwed, interestRate)
+        tempAmountOwed += interest - monthlyPayment
+      }
+    }
+
+    return monthsToPayOffLoan
+  }
+
+  calculate() {
+    const max = (x, y) => (x > y ? x : y)
+    const maxMonthsToPayOffLoan = this.state.loans
+      .map(loan => loan.monthsToPayOffLoan)
+      .reduce(max)
```

Comment:
yeah maybe we should remove the state update from `calculate` and have a method called `handleCalculate` that calls this and does the state update below. then we can use the side-effect-free `calculate` above in `handleChange`, too.